java/java.lsp.server/src/org/netbeans/modules/java/lsp/server/protocol/Server.java

```diff
@@ -948,8 +949,14 @@ private void collectProjectCandidates(FileObject fo, List<FileObject> candidates
 
         private void initializeOptions() {
             getWorkspaceProjects().thenAccept(projects -> {
+                ConfigurationItem item = new ConfigurationItem();
+                item.setSection(NETBEANS_JAVA_HINTS);
+                client.configuration(new ConfigurationParams(Collections.singletonList(item))).thenAccept(c -> {
+                    if (c != null && !c.isEmpty() && c.get(0) instanceof JsonObject) {
```

Comment:
What will happen if the setting does not exist at all? Will the diagnostics be blocked forever (due to the `hintsPrefStatus`)? I think that if the setting does not exist, the diagnostic computation needs to be unblocked as well.

(Not sure how difficult is to write a test for that, but it might be a good idea to write a test for it.)